src/newwave/entity.mo

```diff
@@ -17,88 +17,275 @@ import Result "mo:base/Result";
 import Random "mo:base/Random";
 import Blob "mo:base/Blob";
 
-import EntityType "entity_type";
-import EntitySettings "entity_settings";
-
 import Utils "Utils";
-
-// example to include functions on type object:
-  // type Counter = { inc : () -> Nat };
-  /* func Counter() : { inc : () -> Nat } =
-  object {
-    var c = 0;
-    public func inc() : Nat { c += 1; c }
-  }; */
+import Text "mo:base/Text";
+import Types "Types";
 
 module {
 
+  /**
+   * Types of errors for finding the attached bridges
+  */
+  public type EntityAttachedBridgesErrors = {
+    #EntityNotFound;
+    #Error;
+  };
+
+  /**
+   * Return type for when finding the ids of attached bridges
+  */
+  public type EntityAttachedBridges = Types.Result<[Text], EntityAttachedBridgesErrors>;
+
+  /**
+   * Defines the errors for the public API when trying to retrieve an Entity ID
+  */
+  public type EntityIdErrors = {
+    #Unauthorized;
+    #EntityNotFound;
+    #Error;
+  };
+
+  /**
+   * Defines the result type for trying to retieve the ID of an entity from
+   * the public API
+  */
+  public type EntityIdResult = Types.Result<Text, EntityIdErrors>;
+
+  /**
+   * Defines the entity errors that can occur when trying to retrieve an entity
+  */
+  public type EntityErrors = {
+    #Unauthorized : Text;
+    #EntityNotFound;
+    #Error;
+  };
+
+  /**
+   * Defines the result type for trying to retrieve an Entity from
+   * the public API
+  */
+  public type EntityResult = Types.Result<Entity, EntityErrors>;
+
+  /**
+   * Stores entity specific settings
+  */
+  public class EntitySettings() {
+    var mainSetting : Text = "default";
+  };
+
+  /**
+   * The available entity types that can be used to describe an entity
+  */
+  public type EntityType = {
+    #BridgeEntity;
+    #Webasset;
+    #Person;
+    #Location;
+  };
+
+  /**
+   * Type that defines the attributes for an Entity
+  */
   public type Entity = {
-    internalId : Text;
+    /**
+     * The ID of the Entity that is used to store it in
+     * in the entity database
+    */
+    id : Text;
```

Comment:
we can use a BaseEntity type for the attributes that are shared across Entity and Bridge